### DISCUSSION TITLE
JSON parse error on valid JSON

### DISCUSSION THREAD

**Discussion by DanySK** [renovatebot/renovate#18090] [Request Help] [answered]
I believe I am hitting a bug in Renovate.
I had the following configuration that worked flawlessly for months:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "github>DanySK/renovate-config"
  ],
  "packageRules": [
    {
      "description": "Trigger a minor update on some dependencies",
      "matchPackageNames": [
        "org.apache.commons:commons-math3",
        "org.danilopianini:listset"
      ],
      "semanticCommitScope": "api-deps"
    },
    {
      "description": "Dokka updates should have type docs",
      "matchPackageNames": [
        "org.jetbrains.dokka",
      ],
      "semanticCommitType": "docs"
    },
    {
      "description": "Trigger a patch update on some dependencies",
      "matchPackageNames": [
        "org.jetbrains.kotlin.jvm",
        "org.protelis:protelis-interpreter",
        "org.protelis:protelis-lang"
      ],
      "matchPackagePrefixes": [
        "it.unibo.scafi"
      ],
      "semanticCommitScope": "core-deps"
    }
  ]
}
```

out of the blue, renovate quit working a week ago, with the following error in the dashboard:

```
INFO: Repository has invalid config
{
  "error": {
    "validationSource": "renovate.json",
    "validationError": "Invalid JSON (parsing failed)",
    "validationMessage": "Syntax error near      ],\n  ",
    "message": "config-validation",
    "stack": "Error: config-validation\n    at checkForRepoConfigError (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/merge.js:153:19)\n    at mergeRenovateConfig (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/merge.js:175:5)\n    at async getRepoConfig (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/config.js:15:14)\n    at async initRepo (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/index.js:33:14)\n    at async Object.renovateRepository (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/index.js:39:18)\n    at async renovateRepository (/home/ubuntu/renovateapp/app/worker/index.js:264:26)\n    at async /home/ubuntu/renovateapp/app/worker/index.js:458:5"
  }
}
```

However, I validated the file, and it looks like it's compliant with RFC 8259. Can you help me understand where the mistake is, or whether it is a bug in Renovate?

Exact file: https://github.com/AlchemistSimulator/Alchemist/blob/0b0fce3daa3f3e9ed34a065af5099e3b5a8ccaf1/renovate.json

**Comment by rarkins** [comment, marked as answer]
This part is invalid JSON:

```
      "matchPackageNames": [
        "org.jetbrains.dokka",
      ],
```

I thought @viceice refactored our config parsing to use JSON5 under the hood though. I'm not sure what may have changed recently so that JSON is being used to parse instead, but the above snippet would explain what it is and if you remove the trailing comma you should be back up and running.

**Reply by viceice** [reply]
strange. somebody needs to debug this

**Reply by DanySK** [reply]
> This part is invalid JSON:
> 
> ```
>       "matchPackageNames": [
>         "org.jetbrains.dokka",
>       ],
> ```
> 
> I thought @viceice refactored our config parsing to use JSON5 under the hood though. I'm not sure what may have changed recently so that JSON is being used to parse instead, but the above snippet would explain what it is and if you remove the trailing comma you should be back up and running.

Removal of the trailing comma worked.

**Reply by viceice** [reply]
@DanySK are you using the GitHub app or any other self-hosted platform?

**Reply by DanySK** [reply]
GitHub app

**Reply by viceice** [reply]
@rarkins it seems this line is reading the file as plain JSON and fails.

https://github.com/renovatebot/renovate/blob/cbbb74e6fedb7597c25d19c872909cca5ff28bc5/lib/workers/repository/init/merge.ts#L130

**Reply by viceice** [reply]
we probably can remove that validation?

**Reply by rarkins** [reply]
And not this one earlier? https://github.com/renovatebot/renovate/blob/cbbb74e6fedb7597c25d19c872909cca5ff28bc5/lib/workers/repository/init/merge.ts#L117-L119

This validator was used previously because it was better than JSON.parse at identifying the failure

**Reply by viceice** [reply]
ups, didn't seen the duplicate validation. let's check the json5 parser error. maybe its good enough.
